cmdlets/sqlserver/New-MiLink.ps1

```diff
@@ -0,0 +1,318 @@
+        Write-Verbose "Import MI DBM endpoint certificate into SQL Server [started]"
+        $instanceDBMCertificate = $instance | Get-AzSqlInstanceEndpointCertificate -Name "DATABASE_MIRRORING"
+        $instanceDBMCertificatePublicKey = $instanceDBMCertificate.PublicKey
+        $queryInstanceDBMCertificateImport = "CREATE CERTIFICATE [$miFQDN] FROM BINARY = $instanceDBMCertificatePublicKey"
+        $miCertInSqlServer = Invoke-Sqlcmd -Query "Select * from sys.certificates where name = N'$miFQDN'"
+        if (!$miCertInSqlServer) {
+            Invoke-Sqlcmd -query $queryInstanceDBMCertificateImport -ServerInstance $SqlInstance
+        }
+        Write-Verbose "Import MI DBM endpoint certificate into SQL Server [completed]"
+        # TODO: should we also check for expiration date etc? Or publickey Mismatch? or any other case for which we'd wanna clean up?
+        #Invoke-Sqlcmd -query "DROP CERTIFICATE [$miFQDN]" -ServerInstance $SqlInstance  
+        #Invoke-Sqlcmd -query $queryInstanceDBMCertificateImport -ServerInstance $SqlInstance
+
+        # Fetch the public key of SQL Server authentication certificate (outputs a binary key)
+        Write-Verbose "Export SQL server DBM endpoint certificate to MI [started]"
+        $serverTrustCertificatePK = $dbmCert.EndpointCertificatePublicKey
+        $hexCert = [System.Text.StringBuilder]::new($serverTrustCertificatePK.Length * 2)
+        foreach ($byte in $serverTrustCertificatePK) {
+            $hexCert.AppendFormat("{0:x2}", $byte) | Out-Null
+        }
+        $dbmCert = "0x" + ($hexCert.ToString())
+        # we should be able to use sp_get_endpoint_cert where type=4 to get the cert we need to push to MI ($dbmCert)
+        # import the public key of authentication certificate from SQL Server to Managed Instance
+        $boxCertName = $boxCertName ? $boxCertName : "DBM_Certificate_${SqlInstance}" # this naming is pretty random, TODO: come up with something better
+        $miServerTrustCerts = $instance | Get-AzSqlInstanceServerTrustCertificate
+        [Func[object, bool]] $delegate = { param($c); return ($c.Name -eq $boxCertName) -or ($dbmCert -match $c.PublicKey) }
+        $matchedCert = [Linq.Enumerable]::Where($miServerTrustCerts, $delegate)
+        if (!$matchedCert) {
+            New-AzSqlInstanceServerTrustCertificate -InstanceObject $instance -CertificateName $boxCertName -PublicKey $dbmCert
+        }
+        Write-Verbose "Export SQL server DBM endpoint certificate to MI [completed]"
+
+        #
+        Write-Verbose "Create availability groups [started]"
+        $queryAG =
+        @"
+CREATE AVAILABILITY GROUP [$PrimaryAvailabilityGroup]
+WITH (CLUSTER_TYPE = NONE)
+FOR DATABASE [$DatabaseName]
+REPLICA ON N'chimera' WITH (ENDPOINT_URL = N'$sourceEndpoint', FAILOVER_MODE = MANUAL, AVAILABILITY_MODE = SYNCHRONOUS_COMMIT, SEEDING_MODE = AUTOMATIC, SECONDARY_ROLE(ALLOW_CONNECTIONS = ALL));
```

Comment:
What is 'chimera'?

Reply:
good catch, that should be `$SqlInstance`